Gems/GradientSignal/Code/Source/Editor/EditorGradientImageCreatorUtils.cpp

```diff
@@ -0,0 +1,219 @@
+            return 2;
+        case OutputFormat::R32:
+            return 4;
+        default:
+            AZ_Assert(false, "Unsupported output image format (%d)", format);
+            return 0;
+        }
+    }
+
+    AZStd::vector<AZ::u8> CreateDefaultImageBuffer(int imageResolutionX, int imageResolutionY, int channels, OutputFormat format)
+    {
+        // Fill in our image buffer. Default all values to 0 (black)
+        int bytesPerChannel = ImageCreatorUtils::GetBytesPerChannel(format);
+        const size_t imageSize = imageResolutionX * imageResolutionY * channels * bytesPerChannel;
+        AZStd::vector<AZ::u8> pixels(imageSize, 0);
+
+        // If we're saving a 4-channel image, loop through and set the Alpha channel to opaque.
+        if (channels == 4)
+        {
+            for (size_t alphaIndex = (channels - 1); alphaIndex < (imageResolutionX * imageResolutionY * channels); alphaIndex += channels)
+            {
+                switch (format)
+                {
+                case OutputFormat::R8:
+                    {
+                        pixels[alphaIndex] = std::numeric_limits<AZ::u8>::max();
+                        break;
+                    }
+                case OutputFormat::R16:
+                    {
+                        auto actualMem = reinterpret_cast<AZ::u16*>(pixels.data());
+                        actualMem[alphaIndex] = std::numeric_limits<AZ::u16>::max();
+                        break;
+                    }
+                case OutputFormat::R32:
+                    {
+                        auto actualMem = reinterpret_cast<float*>(pixels.data());
+                        actualMem[alphaIndex] = 1.0f;
+                        break;
+                    }
+                }
+            }
+        }
+
+        return pixels;
+    }
+
+
+    bool WriteImage(
+        const AZStd::string& absoluteFileName,
+        int imageResolutionX,
+        int imageResolutionY,
+        int channels,
+        OutputFormat format,
+        const AZStd::span<const AZ::u8>& pixelBuffer,
+        bool showProgressDialog)
+    {
+        OIIO::TypeDesc pixelFormat = OIIO::TypeDesc::UINT8;
+        switch (format)
+        {
+        case OutputFormat::R8:
+            pixelFormat = OIIO::TypeDesc::UINT8;
+            break;
+        case OutputFormat::R16:
+            pixelFormat = OIIO::TypeDesc::UINT16;
+            break;
+        case OutputFormat::R32:
+            pixelFormat = OIIO::TypeDesc::FLOAT;
+            break;
+        default:
+            AZ_Assert(false, "Unsupported output image format (%d)", format);
+            return false;
+        }
+
+        QProgressDialog* saveDialog = nullptr;
+
+        // Show a dialog box letting the user know the image is being written out.
+        // For large image sizes, it can take 15+ seconds to create and write out the image.
+        if (showProgressDialog)
+        {
+            saveDialog = new QProgressDialog();
```

Comment:
Could parent this to `AzToolsFramework::GetActiveWindow()` so that it will get auto-centered on whichever window the user is active on.